## Project - France

In [1]:
import pandas as pd
import numpy as np
import plotly_express as px
regions = pd.read_csv("./Data/noc_regions.csv")
athlete_events = pd.read_csv("./Data/athlete_events.csv")

In [2]:
regions.head(3)

NOC       region                 notes
0  AFG  Afghanistan                   NaN
1  AHO      Curacao  Netherlands Antilles
2  ALB      Albania                   NaN

- a. hur många länder som är med i datan?

In [3]:
print(f"{regions['NOC'].nunique() = }")
print(f"{regions['region'].nunique() = }")
print(f"{athlete_events['NOC'].nunique() = }")
print(f"{athlete_events['Team'].nunique() = }")

regions['NOC'].nunique() = 230
regions['region'].nunique() = 206
athlete_events['NOC'].nunique() = 230
athlete_events['Team'].nunique() = 1184


Svar: Det finns 206 'region' representerade i datasetet och det finns 230 "NOC"'s (National Olympic Committée's) representerade. Att det är olika antal beror på att samma 'region' historiskt har haft flera olika nationella olympiska komittéer.
Några exepel:
- 'region' som heter 'Germany' har historiskt varit representerade av flera NOC's: Väst Tyskland (FRG), Östtyskland (GDR) och Saar Protectorate (SAA).
- 'region' som heter 'Russia' har historiskt varit representerade av flera NOC's: Soviet Union (URS), Unified Team (EUN).
- Det finns 1184 'Team' representerade i datasetet. 
  
Slutsats: Man kan säga att det finns 230 länder representerade, eftersom det finns 230 NOC's representerade. Men det är en definitionsfråga.
  
[Link to Wikipedia](https://en.wikipedia.org/wiki/All-time_Olympic_Games_medal_table)

In [4]:
display(regions[regions['region'] == 'Germany'])
display(regions[regions['region'] == 'Russia'])

NOC   region notes
70   FRG  Germany   NaN
76   GDR  Germany   NaN
79   GER  Germany   NaN
173  SAA  Germany   NaN

NOC  region notes
66   EUN  Russia   NaN
171  RUS  Russia   NaN
214  URS  Russia   NaN

In [5]:
athlete_events.head(3)


ID                 Name Sex   Age  Height  Weight     Team  NOC  \
0   1            A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   2             A Lamusi   M  23.0   170.0    60.0    China  CHN   
2   3  Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   

         Games  Year  Season       City       Sport  \
0  1992 Summer  1992  Summer  Barcelona  Basketball   
1  2012 Summer  2012  Summer     London        Judo   
2  1920 Summer  1920  Summer  Antwerpen    Football   

                          Event Medal  
0   Basketball Men's Basketball   NaN  
1  Judo Men's Extra-Lightweight   NaN  
2       Football Men's Football   NaN

- b. vilka länder är med? (förkortningarna räcker)

In [6]:
unique_regions = regions['region'].unique() # skapar en variabel ifall vi behöver den senare.
NOCs = pd.Index(regions['NOC'].unique())
print(f"Dessa National Olympic Commités är med i datasetet: {NOCs}.")

Dessa National Olympic Commités är med i datasetet: Index(['AFG', 'AHO', 'ALB', 'ALG', 'AND', 'ANG', 'ANT', 'ANZ', 'ARG', 'ARM',
       ...
       'VIE', 'VIN', 'VNM', 'WIF', 'YAR', 'YEM', 'YMD', 'YUG', 'ZAM', 'ZIM'],
      dtype='object', length=230).


-  c. vilka sporter är med?

In [7]:
sports = pd.Index(athlete_events['Sport'].sort_values().unique())
print(f"Dessa sporter är med: {sports}.")

Dessa sporter är med: Index(['Aeronautics', 'Alpine Skiing', 'Alpinism', 'Archery',
       'Art Competitions', 'Athletics', 'Badminton', 'Baseball', 'Basketball',
       'Basque Pelota', 'Beach Volleyball', 'Biathlon', 'Bobsleigh', 'Boxing',
       'Canoeing', 'Cricket', 'Croquet', 'Cross Country Skiing', 'Curling',
       'Cycling', 'Diving', 'Equestrianism', 'Fencing', 'Figure Skating',
       'Football', 'Freestyle Skiing', 'Golf', 'Gymnastics', 'Handball',
       'Hockey', 'Ice Hockey', 'Jeu De Paume', 'Judo', 'Lacrosse', 'Luge',
       'Military Ski Patrol', 'Modern Pentathlon', 'Motorboating',
       'Nordic Combined', 'Polo', 'Racquets', 'Rhythmic Gymnastics', 'Roque',
       'Rowing', 'Rugby', 'Rugby Sevens', 'Sailing', 'Shooting',
       'Short Track Speed Skating', 'Skeleton', 'Ski Jumping', 'Snowboarding',
       'Softball', 'Speed Skating', 'Swimming', 'Synchronized Swimming',
       'Table Tennis', 'Taekwondo', 'Tennis', 'Trampolining', 'Triathlon',
       'Tug-Of-War', 'V

-  d. vilka medaljtyper finns det?

In [8]:
medal_types = pd.Index(athlete_events['Medal'].unique())
medal_types = medal_types.dropna()
print(f'The medal types are {medal_types[0]}, {medal_types[1]} and {medal_types[2]}.')

The medal types are Gold, Bronze and Silver.


-  e. ta reda på statistik för åldern: medelvärde, median, min, max, standardavvikelse,

In [9]:
# age_statistics = athlete_events["Age"].agg(["mean","median","min","max","std"])
# age_statistics
athlete_events['Age'].describe()

count    261642.000000
mean         25.556898
std           6.393561
min          10.000000
25%          21.000000
50%          24.000000
75%          28.000000
max          97.000000
Name: Age, dtype: float64

### f. utforska datan vidare med egna frågor.
- Vilket land har fått flest guldmedaljer
- 10 bästa länderna (om hinner: uppdelat på sommar och vinter OS) (Arsen)

In [10]:
# 0f, kod här:

### 0f. Om vi har tid:
- Karta med alla länder som varit med i sommar-OS och vinter-OS.

In [11]:
# 0f, kartor, kod här:

### g: Diagram över könsfördelningen (Daniel)
- könsfördelning per OS över tid

In [32]:
# 0g kod här:
#x = athlete_events[athlete_events['Sex'] == 'M'].value_counts()
x = athlete_events['Sex'].value_counts()
x = pd.DataFrame(x)
x.reset_index(inplace=True)
# x
fig = px.bar(x, x='Sex', y='count', template="ggplot2")
fig.show()
#px.bar(athlete_events)

###  h. diagram över topp 10 länder som tagit flest medaljer:
- Arsen

In [13]:
# 0h kod här:

### i. plotta gärna fler saker som är intressant:

In [14]:
# 0i kod här: